src/blockfilter.cpp

```diff
@@ -193,3 +195,41 @@ bool GCSFilter::MatchAny(const std::set<Element>& elements) const
     const std::vector<uint64_t> queries = BuildHashedSet(elements);
     return MatchInternal(queries.data(), queries.size());
 }
+
+static std::set<GCSFilter::Element> BasicFilterElements(const CBlock& block,
```

Comment:
In commit "blockfilter: Construction of basic block filters." (f7a5bdb54ba597fe06bcb54e886c43436899ab48)

I think it could be beneficial if `GCSFilter` defined: 

```c++
typedef std::unordered_set<Element> Elements;
```

And if this function, the `GCSFilter` constructor, `BuildHashedSet`, and `MatchAny` all used  `GCSFilter::Elements` instead of `std::set<GCSFilter::Element>`. It'd be nice to decrease the coupling to one data structure, and since `std::set` is an ordered set, these functions are pointlessly also sorting the elements when order is not important.

Reply:
I added the typedef to `std::set`. In commit e645d53b3d5fa5912e3a99cc4fec74aee64e81e5, I change it to `std::unordered_set` which uses a randomly selected SipHash instance to hash the byte vector. Let me know if you think this is acceptable or whether it should be left as `std::set`.